### PR TITLE
Make memcached protocol and locator configurable.

### PR DESCRIPTION
**Pull request by korablevalexander**
It's important for mcrouter users, because mcrouter doesn't support binary protocol - https://github.com/facebook/mcrouter/issues/6

### CODE REVIEW

File: docs/content/configuration/caching.md [hunk 1]
```diff
@@ -84,7 +84,8 @@ Uses memcached as cache backend. This allows all nodes to share the same cache.
 |`druid.cache.maxObjectSize`|Maximum object size in bytes for a Memcached object.|52428800 (50 MB)|
 |`druid.cache.memcachedPrefix`|Key prefix for all keys in Memcached.|druid|
 |`druid.cache.numConnections`|Number of memcached connections to use.|1|
-
+|`druid.cache.protocol`|Memcached communication protocol. Can be binary or text.|binary|
+|`druid.cache.locator`|Memcached locator. Can be consistent or array_mod.|consistent|
```
Comment (nishantmonu51) [inline docs/content/configuration/caching.md]: Valid values - consistent, array_mod and vbucket.

**Comment by nishantmonu51** [comment]
@korablevalexander please fix travis failures,
failing to compile currently - 
[ERROR] /home/travis/build/druid-io/druid/server/src/main/java/io/druid/client/cache/MemcachedCache.java:72: Using a static member import should be avoided - io.druid.java.util.common.StringUtils.toUpperCase. [AvoidStaticImport]

File: docs/content/configuration/caching.md (hunk 1 above)
Comment (korablevalexander) [reply docs/content/configuration/caching.md]: https://github.com/druid-io/druid/blob/c0beb78ffdb5807ceea8a340838eab01d6a18d1d/server/src/main/java/io/druid/client/cache/MemcachedCustomConnectionFactoryBuilder.java#L62

Comment (korablevalexander) [reply docs/content/configuration/caching.md]: For native spymemcached `ConnectionFactoryBuilder` it's invalid too.

Comment (nishantmonu51) [reply docs/content/configuration/caching.md]: cool, thanks.

**Comment by korablevalexander** [comment]
This flap is not a my fault https://travis-ci.org/druid-io/druid/jobs/347423481. Restart plz.

**Comment by drcrallen** [comment]
ah shoot, I forgot to check cla first, checking

**Comment by drcrallen** [comment]
@korablevalexander would you mind filling out http://druid.io/community/cla.html please?

**Comment by korablevalexander** [comment]
Done)

**Comment by drcrallen** [comment]
Thanks!

**Comment by korablevalexander** [comment]
Hi, @nishantmonu51, i'm a little confused that this changes not in 0.12.0. I had to make a pr in 0.12.0? When will master be mergerd into version branch.

**Comment by nishantmonu51** [comment]
This will land in 0.13.0 release.

**Pull request merged**